frame/lending/src/mocks/mod.rs

```diff
@@ -432,31 +436,47 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
 
 	let mut ext = sp_io::TestExternalities::new(storage);
 	ext.execute_with(|| {
-		System::set_block_number(0);
+		System::set_block_number(1);
 		Timestamp::set_timestamp(MILLISECS_PER_BLOCK);
 	});
 	ext
 }
 
-/// Progress to the given block, and then finalize the block.
-#[allow(dead_code)]
-pub fn run_to_block(n: BlockNumber) {
-	Lending::on_finalize(System::block_number());
-	for b in (System::block_number() + 1)..=n {
-		next_block(b);
-		if b != n {
-			Lending::on_finalize(System::block_number());
-		}
-	}
-}
-
-pub fn process_block(n: BlockNumber) {
-	next_block(n);
-	Lending::on_finalize(n);
-}
-
-pub fn next_block(n: u64) {
-	System::set_block_number(n);
-	Lending::on_initialize(n);
-	Timestamp::set_timestamp(MILLISECS_PER_BLOCK * n);
+// BLOCK HELPERS
+
+// /// Progress to the given block, and then finalize the block.
+// pub fn run_to_block(n: BlockNumber) {
+// 	Lending::on_finalize(System::block_number());
+// 	for b in (System::block_number() + 1)..=n {
+// 		next_block(b);
+// 		if b != n {
+// 			Lending::on_finalize(System::block_number());
+// 		}
+// 	}
+// }
+
+/// Processes the specified amount alls [`next_block()`] and then calls
+/// [`Lending::finalize`](OnFinalize::on_finalize).
+pub fn process_and_progress_blocks(blocks_to_process: usize) {
+	std::iter::repeat(())
```

Comment:
nice, intrestign how it works. so i would prefer if possible 

(0..blocks_to_process).for_each